packages/app-store/office365calendar/api/callback.ts

```diff
@@ -74,6 +76,50 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     },
   });
 
+  // Set the isDefaultCalendar as selectedCalendar
+  // If a user has multiple calendars, keep on making calls until we find the default calendar
+  let defaultCalendar: OfficeCalendar | undefined = undefined;
+  let requestUrl = "https://graph.microsoft.com/v1.0/me/calendars?$select=id,isDefaultCalendar";
+  let finishedParsingCalendars = false;
+
+  while (!finishedParsingCalendars) {
+    const calRequest = await fetch(requestUrl, {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${responseBody.access_token}`,
+        "Content-Type": "application/json",
+      },
+    });
+    let calBody = await handleErrorsJson<{ value: OfficeCalendar[]; "@odata.nextLink"?: string }>(calRequest);
+
+    if (typeof responseBody === "string") {
+      calBody = JSON.parse(responseBody) as { value: OfficeCalendar[] };
+    }
+
+    const findDefaultCalendar = calBody.value.find((calendar) => calendar.isDefaultCalendar);
+
+    if (findDefaultCalendar) {
+      defaultCalendar = findDefaultCalendar;
+    }
+
+    if (calBody["@odata.nextLink"]) {
+      requestUrl = calBody["@odata.nextLink"];
+    } else {
+      finishedParsingCalendars = true;
+    }
+  }
+
+  if (defaultCalendar?.id && req.session?.user?.id) {
+    await prisma.selectedCalendar.create({
+      data: {
+        userId: req.session?.user.id,
+        integration: "office365_calendar",
+        externalId: defaultCalendar.id,
+        credentialId: credential.id,
+      },
+    });
+  }
```

Comment:
`suggestion-non-blocking`: I'd prefer encapsulating this logic into its own function, and only running it if the actual `req.session.user.id` exists, if not running it is useless no?

```ts
const findAndStoreDefaultCalendar = async (accessToken: string, credentialId: string, userId: string) => {
  let requestUrl = "https://graph.microsoft.com/v1.0/me/calendars?$select=id,isDefaultCalendar";
  
  try {
    while (true) {
      const response = await fetch(requestUrl, {
        method: "GET",
        headers: {
          Authorization: `Bearer ${accessToken}`,
          "Content-Type": "application/json",
        },
      });

      const { value: calendars, "@odata.nextLink": nextLink } = await handleErrorsJson<{ value: OfficeCalendar[]; "@odata.nextLink"?: string }>(response);
      const defaultCalendar = calendars.find(calendar => calendar.isDefaultCalendar);

      if (defaultCalendar) {
        await prisma.selectedCalendar.create({
          data: {
            userId,
            integration: "office365_calendar",
            externalId: defaultCalendar.id,
            credentialId,
          },
        });

        break;
      }

      if (!nextLink) {
        break;
      }

      requestUrl = nextLink;
    }
  } catch (error) {
    throw error;
  }
}

if (req.session?.user?.id) {
  await findAndStoreDefaultCalendar(responseBody.access_token, credential.id, req.session.user.id);
}
```

Reply:
Even better, looking at the entire code it looks like the credential create will also fail if there is no session user, so the ideal fix is to return early out of the entire callback when the session user id isn't truthy.